scene/main/node.cpp

```diff
@@ -2857,6 +2857,9 @@ void Node::_bind_methods() {
 	BIND_CONSTANT(NOTIFICATION_INTERNAL_PHYSICS_PROCESS);
 	BIND_CONSTANT(NOTIFICATION_POST_ENTER_TREE);
 
+	BIND_CONSTANT(NOTIFICATION_EDITOR_PRE_SAVE);
+	BIND_CONSTANT(NOTIFICATION_EDITOR_POST_SAVE);
```

Comment:
This code is required because the editor plugin is not properly used.

Reply:
This code is required for saving the Bone2D nodes correctly when IK is being executed in the editor, it doesn't involve the editor plugin or anything. It is to keep one of the fundamental promises of IK: that the base pose will *not* be modified by the IK, allowing you to return to it when you turn off/disable IK. Without these notifications, saving the scene in the editor with the Bone2D pose enabled will incorrectly save the IK pose.

These notifications are required for that reason.